### PR TITLE
Remove deprecated as_messages function.

### PR DESCRIPTION
**Pull request by bjlittle**

### CODE REVIEW

File: iris_grib/__init__.py [hunk 1]
```diff
@@ -1065,22 +1065,6 @@ def save_pairs_from_cube(cube):
         yield (slice2D, grib_message)
 
 
-def as_messages(cube):
-    """
-    .. deprecated:: 1.10
```
Comment (dkillick) [inline iris_grib/__init__.py]: Again with the deprecated in Iris 1.10 question...

File: iris_grib/__init__.py [hunk 2]
```diff
@@ -1024,7 +1024,7 @@ def save_grib2(cube, target, append=False, **kwargs):
     See also :func:`iris.io.save`.
 
     """
-    messages = as_messages(cube)
+    messages = (message for _, message in save_pairs_from_cube(cube))
```
Comment (dkillick) [inline iris_grib/__init__.py]: Given that you've changed the behaviour of this function, have you also updated the tests for it?

**Comment by dkillick** [comment]
See also: #8.

**Comment by dkillick** [comment]
@bjlittle guess what? more merge conflicts...

**Comment by bjlittle** [comment]
@dkillick I'll rebase after #30 lands ...

**Comment by dkillick** [comment]
#30 has landed!

File: iris_grib/__init__.py (hunk 1 above)
Comment (bjlittle) [inline iris_grib/__init__.py]: @dkillick You now know that we're agreed and happy to deprecate :wink:

File: iris_grib/__init__.py (hunk 2 above)
Comment (bjlittle) [inline iris_grib/__init__.py]: @dkillick As you can see, there is [no test coverage](https://coveralls.io/builds/6731708/source?filename=iris_grib%2F__init__.py#L996) for `save_grib2` ... tsk, tsk, tsk!

Guess I'd better create some :wink:

**Comment by bjlittle** [comment]
@dkillick @marqh

**Comment by marqh** [comment]
looking good to me

**Pull request merged**